src/utils.rs

```diff
@@ -425,15 +422,13 @@ pub struct Signature {
     pub signature: Vec<u8>,
 }
 
-impl TryFrom<TPMT_SIGNATURE> for Signature {
-    type Error = Error;
-
-    fn try_from(tss_signature: TPMT_SIGNATURE) -> Result<Self> {
+impl Signature {
+    pub unsafe fn try_from(tss_signature: TPMT_SIGNATURE) -> Result<Self> {
```

Comment:
I think that is a very good decision to put the `unsafe` on the method here as we can not trust that the `tss_signature` argument is always well-composed. The caller of this method needs to make the decision as only him knows where the parameter comes from.

Reply:
I was thinking: "what if the method is not `public`?"
But even so, it would maybe be wrong to put it safe anyway.

Reply:
Maybe a comment would be nice to explain those things? Or that could be part of the documentation of the method itself.

Reply:
Yeah, I decided to just leave it to documentation, a comment isn't as visible for users